src/sqlfluff/rules/ambiguous/AM08.py

```diff
@@ -0,0 +1,154 @@
+"""Implementation of Rule AM08."""
+
+from typing import Optional, Tuple
+
+from sqlfluff.core.parser import BaseSegment, KeywordSegment, WhitespaceSegment
+from sqlfluff.core.rules import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
+
+
+class Rule_AM08(BaseRule):
+    """Implicit cross join detected.
+
+    **Anti-pattern**
+
+    Cross joins are valid, but rare in the wild - and more often created by mistake
+    than on purpose. This rule catches situations where a cross join has been specified,
+    but not explicitly and so the risk of a mistaken cross join is highly likely.
```

Comment:
What would we think of also saying something like:
>Cross joins can even cause query failures with large datasets, because they generate _all possible combinations_ of the two input tables, which can be a **huge** number of records.

I suggest this because cross join is a pretty technical term that some users won't be familiar with (for the very reason that they are rare), so this is an opportunity to provide some education and context for the existence of the rule.